### PR TITLE
Release v1.0.6

### PR DESCRIPTION
**Pull request by alinmigea**
@plentymarkets/team-multichannel-koboldmaki

### CODE REVIEW

File: resources/lang/de/log.properties [hunk 1]
```diff
@@ -0,0 +1,11 @@
+variationNotPartOfExportStock       = "Variante ohne Bestand."
```
Comment (mbrueschke) [inline resources/lang/de/log.properties]: you can delete the log properties we dont need anymore :)

File: resources/lang/en/log.properties [hunk 2]
```diff
@@ -0,0 +1,11 @@
+variationNotPartOfExportStock       = "Variation has no stock."
+variationNotPartOfExportPrice       = "Variation price is negative."
```
Comment (mbrueschke) [inline resources/lang/en/log.properties]: you can delete the log properties we dont need anymore :)

File: src/Generator/ShoppingCOM.php [hunk 3]
```diff
@@ -40,113 +64,267 @@ public function __construct(ArrayHelper $arrayHelper)
...
+
+        $this->elasticExportPropertyHelper = pluginApp(ElasticExportPropertyHelper::class);
+
+        $settings = $this->arrayHelper->buildMapFromObjectList($formatSettings, 'key', 'value');
+
+        $this->setDelimiter(self::DELIMITER);
+
+        $this->addCSVContent($this->head());
+
+        $startTime = microtime(true);
```
Comment (mbrueschke) [inline src/Generator/ShoppingCOM.php]: the microtime is not necessary  anymore

File: src/Generator/ShoppingCOM.php [hunk 4]
```diff
@@ -40,113 +64,267 @@ public function __construct(ArrayHelper $arrayHelper)
...
 
-            //Create a List of all VariationIds
-            $variationIdList = array();
-            foreach($resultData['documents'] as $variation)
+            do
             {
-                $variationIdList[] = $variation['id'];
-            }
+                // Number of printed lines
+                $this->getLogger(__METHOD__)->debug('ElasticExportShoppingCOM::log.writtenLines', [
```
Comment (mbrueschke) [inline src/Generator/ShoppingCOM.php]: delete log, as we commited

File: src/Generator/ShoppingCOM.php [hunk 5]
```diff
@@ -40,113 +64,267 @@ public function __construct(ArrayHelper $arrayHelper)
...
+                {
+                    break;
+                }
 
-            //Creates an array with the variationId as key to surpass the sorting problem
-            if(isset($idlResultList) && $idlResultList instanceof RecordList)
-            {
-                $this->createIdlArray($idlResultList);
-            }
+                $esStartTime = microtime(true);
```
Comment (mbrueschke) [inline src/Generator/ShoppingCOM.php]: delete microtime, as we commited

File: src/Generator/ShoppingCOM.php [hunk 6]
```diff
@@ -40,113 +64,267 @@ public function __construct(ArrayHelper $arrayHelper)
...
 
-            foreach($resultData['documents'] as $item)
-            {
-                $deliveryCost = $this->elasticExportHelper->getShippingCost($item['data']['item']['id'], $settings);
-                if(!is_null($deliveryCost))
+                // Get the data from Elastic Search
+                $resultList = $elasticSearch->execute();
+
+                $this->getLogger(__METHOD__)->debug('ElasticExportShoppingCOM::log.esDuration', [
```
Comment (mbrueschke) [inline src/Generator/ShoppingCOM.php]: delete log, as we commited

File: src/Generator/ShoppingCOM.php [hunk 7]
```diff
@@ -40,113 +64,267 @@ public function __construct(ArrayHelper $arrayHelper)
...
+                        if($limit == $filter['limit'])
+                        {
+                            $limitReached = true;
+                            break;
+                        }
+
+                        // If filtered by stock is set and stock is negative, then skip the variation
+                        if($this->elasticExportStockHelper->isFilteredByStock($variation, $filter) === true)
+                        {
+                            $this->getLogger(__METHOD__)->info('ElasticExportShoppingCOM::log.variationNotPartOfExportStock', [
```
Comment (mbrueschke) [inline src/Generator/ShoppingCOM.php]: delete log, as we commited

File: src/Generator/ShoppingCOM.php [hunk 8]
```diff
@@ -40,113 +64,267 @@ public function __construct(ArrayHelper $arrayHelper)
...
+
+    /**
+     * Creates the variation row and prints it into the CSV file.
+     *
+     * @param array $variation
+     * @param KeyValue $settings
+     */
+    private function buildRow($variation, KeyValue $settings)
+    {
+        $this->getLogger(__METHOD__)->debug('ElasticExportShoppingCOM::log.variationConstructRow', [
```
Comment (mbrueschke) [inline src/Generator/ShoppingCOM.php]: delete log, as we commited

File: src/Generator/ShoppingCOM.php [hunk 9]
```diff
@@ -40,113 +64,267 @@ public function __construct(ArrayHelper $arrayHelper)
...
+                'Verfügbarkeitdetails' 	=> $this->elasticExportHelper->getAvailability($variation, $settings),
+                'Versand: Landtarif' 	=> $shippingCost,
+                'Produktgewicht'        => $variation['data']['variation']['weightG'],
+                'Produkttyp' 			=> $this->elasticExportPropertyHelper->getItemPropertyByBackendName($variation, 'product_type', $settings->get('lang')),
+                'Grundpreis' 			=> $this->elasticExportPriceHelper->getBasePrice($variation, $priceList['price']),
+            ];
+
+            $this->addCSVContent(array_values($data));
+
+            $this->getLogger(__METHOD__)->debug('ElasticExportShoppingCOM::log.variationConstructRowFinished', [
```
Comment (mbrueschke) [inline src/Generator/ShoppingCOM.php]: delete log, as we commited

File: src/Generator/ShoppingCOM.php [hunk 10]
```diff
@@ -40,113 +64,267 @@ public function __construct(ArrayHelper $arrayHelper)
...
+                'Data row duration' => 'Row printing took: ' . (microtime(true) - $rowTime),
+            ]);
+        }
+        else
+        {
+            $this->getLogger(__METHOD__)->info('ElasticExportShoppingCOM::log.variationNotPartOfExportPrice', [
```
Comment (mbrueschke) [inline src/Generator/ShoppingCOM.php]: delete log, as we commited

File: src/Generator/ShoppingCOM.php [hunk 11]
```diff
@@ -40,113 +64,267 @@ public function __construct(ArrayHelper $arrayHelper)
...
+                if(!isset($this->manufacturerCache) || (isset($this->manufacturerCache) && !array_key_exists($variation['data']['item']['manufacturer']['id'], $this->manufacturerCache)))
                 {
-                    $this->idlVariations[$idlVariation->variationBase->id] = [
-                        'itemBase.id' => $idlVariation->itemBase->id,
-                        'variationBase.id' => $idlVariation->variationBase->id,
-                        'itemPropertyList' => $idlVariation->itemPropertyList,
-                        'variationRetailPrice.price' => $idlVariation->variationRetailPrice->price,
-                        'variationRetailPrice.vatValue' => $idlVariation->variationRetailPrice->vatValue,
-                    ];
+                    $manufacturer = $this->elasticExportHelper->getExternalManufacturerName((int)$variation['data']['item']['manufacturer']['id']);
```
Comment (mbrueschke) [inline src/Generator/ShoppingCOM.php]: we do not need a cache for manufacturer because we already cache it at the eeHelper

**Comment by alinmigea** [comment]
[PropertyHelper PR](https://github.com/plentymarkets/plenty-plugin-elastic-export/pull/159) should be merge before this.

File: src/ResultField/ShoppingCOM.php [hunk 12]
```diff
@@ -79,10 +76,21 @@ public function generateResultFields(array $formatSettings = []):array
...
+
         /**
          * @var LanguageMutator $languageMutator
          */
         $languageMutator = pluginApp(LanguageMutator::class, [[$settings->get('lang')]]);
+
+        /**
+         * @var SkuMutator $skuMutator
+         */
+        $skuMutator = pluginApp(SkuMutator::class);
```
Comment (mbrueschke) [inline src/ResultField/ShoppingCOM.php]: sku wird nicht verwendet

File: src/ResultField/ShoppingCOM.php [hunk 13]
```diff
@@ -140,13 +167,23 @@ public function generateResultFields(array $formatSettings = []):array
...
+                'properties.texts.value',
+                'properties.texts.lang',
+                'properties.valueInt',
+                'properties.valueFloat',
             ],
 
             [
                 $languageMutator,
-                $defaultCategoryMutator
+                $skuMutator,
```
Comment (mbrueschke) [inline src/ResultField/ShoppingCOM.php]: kann entfernt werden

File: src/ResultField/ShoppingCOM.php [hunk 14]
```diff
@@ -162,4 +199,133 @@ public function generateResultFields(array $formatSettings = []):array
...
+    private function getNestedKeyList()
+    {
+        $nestedKeyList['keys'] = [
+            //images
+            'images.all',
+            'images.item',
+            'images.variation',
+
+            //sku
+            'skus',
```
Comment (mbrueschke) [inline src/ResultField/ShoppingCOM.php]: kann entfernt werden

File: src/ResultField/ShoppingCOM.php [hunk 15]
```diff
@@ -162,4 +199,133 @@ public function generateResultFields(array $formatSettings = []):array
...
+                'urlMiddle',
+                'urlPreview',
+                'urlSecondPreview',
+                'url',
+                'path',
+                'position',
+            ],
+
+            //sku
+            'skus' => [
```
Comment (mbrueschke) [inline src/ResultField/ShoppingCOM.php]: kann entfernt werden

**Pull request merged**